ledger/src/entry.rs

```diff
@@ -301,6 +293,34 @@ impl EntrySlice for [Entry] {
         );
         res
     }
+
+    fn verify_tick_hash_count(&self, tick_hash_count: &mut u64, hashes_per_tick: u64) -> bool {
+        // When hashes_per_tick is 0, hashing is disabled.
+        if hashes_per_tick == 0 {
+            return true;
+        }
+
+        for entry in self {
+            *tick_hash_count += entry.num_hashes;
+            if entry.is_tick() {
+                if *tick_hash_count != hashes_per_tick {
+                    warn!(
+                        "invalid tick hash count!: entry: {:#?}, tick_hash_count: {}, hashes_per_tick: {}",
+                        entry,
+                        tick_hash_count,
+                        hashes_per_tick
+                    );
+                    return false;
+                }
+                *tick_hash_count = 0;
+            }
+        }
+        *tick_hash_count <= hashes_per_tick
```

Comment:
I think this: https://github.com/solana-labs/solana/blob/master/ledger/src/poh.rs#L50-L52, enforces that this can be a strict `<`. I think there are tests in poh.rs (might be good to verify) that check the last hash has to come with a tick as well.